File: backend/notifications/src/main/java/com/oxygensend/notifications/config/TwilioProperties.java
package com.oxygensend.notifications.config;

import jakarta.validation.constraints.NotEmpty;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.validation.annotation.Validated;

@Validated
@ConfigurationProperties(prefix = "twilio")
public record TwilioProperties(@NotEmpty
                               String accountSid,
                               @NotEmpty
                               String authToken,
                               @NotEmpty
                               String fromPhoneNumber) {
}

File: backend/notifications/src/main/kotlin/com/oxygensend/notifications/config/TwilioProperties.kt
package com.oxygensend.notifications.config

import jakarta.validation.constraints.NotEmpty
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.validation.annotation.Validated

@Validated
@ConfigurationProperties(prefix = "twilio")
data class TwilioProperties(
    val accountSid: @NotEmpty String,
    val authToken: @NotEmpty String,
    val fromPhoneNumber: @NotEmpty String
)
